experimental/dds/tree2/src/core/rebase/utils.ts

```diff
@@ -9,22 +9,41 @@ import { GraphCommit, mintRevisionTag, mintCommit } from "./types";
 
 /**
  * Contains information about how the commit graph changed as the result of rebasing a source branch onto another target branch.
+ * @example
+ * ```text
+ * Consider the commit graph below containing two branches, X and Y, with head commits C and E, respectively.
+ * Branch Y branches off of Branch X at their common ancestor commit A, i.e. "Y is based off of X at commit A".
+ *
+ *   A ─ B ─ C ← Branch X
+ *   └─ D ─ E ← Branch Y
+ *
+ * Branch Y is then rebased onto Branch X. This results in the following commit graph:
+ *
+ *   A ─ B ─ C ← Branch X
+ *           └─ D'─ E'← Branch Y
+ *
+ * Commits D' and E' are the rebased versions of commits D and E, respectively. This results in:
+ * deletedSourceCommits: [D, E],
+ * rebasedTargetCommits: [B, C],
```

Comment:
This naming is a bit confusing to me. It seems to imply commits that are on the target branch but got rebased. But instead, it actually means commits that the source branch was rebased over?

Reply:
It's a pretty bad name, you are right. I changed it to just `targetCommits`, since `targetCommitsOverWhichSourceCommitsWereRebased` was already taken.